### ISSUE TITLE
No ability to change shipping method when using Payment Request buttons in Chrome

### ISSUE THREAD

**Issue by AJerm** [woocommerce/woocommerce-gateway-stripe#421]
#### What I expected

With Apple pay, there's an option to alter your shipping method in the Apple may modal.

With the new browser payment request buttons it picks one, and doesn't let you change it.

This happens when trying to buy a product on a single product page, also in the checkout. 

The only way I can see to change it is to use the check box options before you click pay, which are only available on the checkout page.

It would be much better to be able to change the method in the browser payment modal, as with picking addresses, cards etc.

The shipping method does appear for the desktop Apple pay in Safari on OSX.

#### Steps to reproduce the issue

You can test it here [link](https://www.cycleprintworks.co.uk/product/cartridge-paper-roll-large-format-printers-fabriano-accademia-24-inch-610mm/)

You may need to click add to basket before the Payment Request button appears. I think there's a clash between the Avada js and maybe your js, blocking it from loading sometimes on the initial page load.

I've already raised a ticket with Avada about this, but the chap I spoke to said as this stripe integration was a beta he couldn't promise anything. I imagine it's a theme issue, but if anyone is able to spot what that issue is too, then that would be cool.

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by roykho** [comment]
Different methods within the same shipping plugin or you're using multiple shipping plugins and can't choose between them?

**Comment by AJerm** [comment]
I'm not using any WooCommerce plugins, just normal WooCommerce and this beta stripe gateway. The stable gateway is also installed, but deactivated.

It looks like WooCommerce services was also installed.  I dont' think I'm using any features from that though. I disabled it and it's the same as above.

It's just two shipping methods, setup in a single zone

![image](https://user-images.githubusercontent.com/1643439/33721606-79beaa62-db5f-11e7-8163-9f6a5a927c4e.png)

**Comment by roykho** [comment]
Ok so you're using multiple shipping methods. I don't think that has been tested yet. I will check. To be clear what I mean is we've tested with a single method say UPS and within it, it gives you different methods of delivery. That is what I mean by we're able to select different ones. In your case, its multiple shipping "company" if you will.

**Comment by roykho** [comment]
I have just tested this in both Apple Pay and Payment Request and in both cases I am able to change the shipping method between flat rate and local pickup. Perhaps you may have something else conflicting in the setup? Have you tried deactivating all plugins except WC and Stripe?

**Comment by AJerm** [comment]
I just tested by disabling all add ons and the issue still arose. 

So I did a clean install, only installed shopfront, woocommerce and the beta stripe integration. 

Set two shipping methods, one paid flat rate and one local collection.

Same issue arises. Apple pay shows a field to edit the the shipping method as so, as intended:

![image](https://user-images.githubusercontent.com/1643439/33736716-8c09b6b8-db8b-11e7-939d-093485544726.png)

Chrome buy now button and modal gives no option to change it, so you're stuck with the first option it picks:

![image](https://user-images.githubusercontent.com/1643439/33736780-c380a1e2-db8b-11e7-9994-2319afb7a1fb.png)

![image](https://user-images.githubusercontent.com/1643439/33736793-d26ef2bc-db8b-11e7-9629-c410f1a8a6ad.png)

You can test it here: https://www.cycleprintworks.co.uk/stripetest/product/test-product/

Admin access at 
user: alex
pass: asdfasdfasdf

**Comment by AJerm** [comment]
I don't know if it's worth noting that Apple Pay takes a few moments to think about what it's doing before it refreshes with the shipping method and associated options. As if it's querying the site to see what shipping methods are available and then updating the screen.

The Chrome modal via the Buy Now button doesn't do anything like that for me

**Comment by roykho** [comment]
@AJerm can you press on the "choose" address button and select an address again and see what happens after.

**Comment by AJerm** [comment]
Ahh okay, so if I select a new address it returns me to the main screen and adds the delivery method field. It's just not appearing on the initial buy now click for some reason

![image](https://user-images.githubusercontent.com/1643439/33759040-b7676e8c-dbf8-11e7-82df-8a45508877dc.png)

**Comment by roykho** [comment]
@AJerm for me on my screen it shows from the start. So perhaps maybe I have more than 1 delivery address set? Maybe you have only 1 so it just defaults to that and there wouldn't be any selection is why? Can you please check?

**Comment by AJerm** [comment]
Okay right, so it is working, but it's a confusing workflow if you have a customer that wants to purchase something for local collection, as they won't want to choose a delivery address. I think this was throwing me.

It looks like when you have lots of addresses (I haven't got a Chrome install to test with no address data), on load the delivery method box is hidden until you manually pick a delivery address.

Only then will the methods appear, and you can change them.

This doesn't really make sense though for many customers who want to collect locally, as it's not intuitive to select a delivery address in order to gain access to the shipping method selector to then tell the checkout you want to collect and not have anything delivered.

As you then have a screen in front of you which kind of contradicts itself. It says you've selected an address you want the product delivered to, then it says you want to collect it.

This is fine on Apple Pay as you don't have to select an address in order to see the box, so you kind of glaze over the address and the only user interaction to specify a different delivery method.

Is there a way to force chrome to preselect the first address in the list in order to force the delivery method box to display on load? Or any way to force the delivery method to display by default on load? 

As I can see the current flow being quite confusing and unintuitive for many people.

**Comment by roykho** [comment]
@AJerm sorry if I wasn't clear, that is not me saying this is not an issue, that was me saying try these steps to see if you're seeing what I am seeing to confirm to me that is where the issue lies. I will definitely look into this but strange it doesn't work the same way as Apple Pay as they're using the same API. So perhaps this is something Stripe needs to fix but I will check this out.

**Comment by AJerm** [comment]
Hey Roykho, no worries. Sorry, my initial last response was a bit longer and I edited it down as I thought I came off a bit harsh. 

If there's some way of forcing a default address so that the shipping method loads by default, or forcing the shipping method to load at the start even without a default address selected that would be great, as it is a little confusing for someone to have to pick a shipping address if they want to collect locally.

They way it works with apple pay is spot on, and such a lovely experience. It would be great if the Chrome version matched it. Perhaps it is a stripe issue.

**Comment by roykho** [comment]
@AJerm ok so I tested this quite a bit and I am not seeing the issue. I removed all the addresses from my Chrome settings except for one so that is the default. When the payment sheet comes up, it still asks to choose a shipping address as first step. It won't let you click on "Pay" without first going through that. But in regards to your question on forcing the only one address to be default, that is a question for Chrome/Google as that is how they designed it to work and not something we can control.

Initially I thought you meant there is a bug with our Stripe plugin where it lets you "Pay" even without selecting the shipping method.

So my suggestion for you is for the products that don't need shipping, just disable shipping for that product so that when they go to pay, shipping address won't even be asked.

**Comment by AJerm** [comment]
It's not that the products don't have shipping, it's that there's also option to collect locally and it doesn't make sense for a user to have to pick a shipping address first, in order to be able to view the option to collect locally as nothing is getting shipped anywhere. 

It's inconsistent with the way Apple Pay works, which is much better in that sense as the shipping method is always visible and never hidden.

**Comment by roykho** [comment]
@AJerm yes I understand what you're saying however we don't have control over that unfortunately.

**Comment by AJerm** [comment]
No worries mate, I thought I'd suggest it anyway.

I encountered a similar issue when purchase something from an ebay seller yesterday.

I wanted to order the item for collection, with no shipping, yet I was forced to manually enter an irrelevant shipping address anyway before I reached the part of the page where I selected that I wished to collect instead.

It seems it's a UX issue with many checkout systems.

It's probably a more common over here in the UK where many shops or online stores are located within a couple of a hours drive from much of the rest of the country, so if we need an item urgently or same day then click and collect is quite important.
